ci/envs/311-latest-conda-forge.yaml

```diff
@@ -26,12 +25,15 @@ dependencies:
   - geodatasets
   # installed in tests.yaml, because not available on windows
   # - postgis
-  - SQLalchemy>=2
-  - psycopg
+  - SQLalchemy<2.1 # test oldest support sqlalchemy version
   - psycopg2
   - libspatialite
   - geoalchemy2
   - pyarrow
   - geoarrow-pyarrow
+  - fiona
   # doctest testing
   - pytest-doctestplus
+  - pip
+  - pip:
+      - git+https://github.com/geopandas/pyogrio.git@main
```

Comment:
Why are you pulling pyogrio from main here?

Reply:
I didn't change this, it's just github doesn't track the renames right (see 7560b8bfcc5). We already test against pyogrio latest in the equivalent on main; https://github.com/geopandas/geopandas/blob/main/ci/envs/310-latest-conda-forge.yaml. I'm not sure why 3.11 makes a difference in terms of installing from source.

Reply:
Let's install pyogrio from conda here. This comes from #2627 and at that time, pyogrio on conda-forge was not available for 3.11.

Reply:
ah makes sense, just pushed a fix for that (it's still probably not a bad idea to understand why this pip install fails but I haven't had a chance to dig into it)